hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/MkdirOperation.java

```diff
@@ -73,11 +79,13 @@ public MkdirOperation(
       final StoreContext storeContext,
       final Path dir,
       final MkdirCallbacks callbacks,
-      final boolean isMagicPath) {
+      final boolean isMagicPath,
```

Comment:
actually, these should be the same flag. so rename it performanceCreation and in s3aFS set to true if the path is magic or `performanceCreation` is true.

Reply:
This comment is from previous patch version.